### ISSUE TITLE
Continue Runs

### ISSUE THREAD

**Issue by cmaerzluft** [carolyner/imabc#14]
Allow for previous run results and objects to be resubmitted to the function if they aren't completed yet.

Multiple phases:
1. Straight restart - nothing has changed between run attempts
2. Target modification - Targets can be added and/or removed between one run and the next
3. Parameter modification - Parameters can be added and/or removed between one run the next

UPDATED: 7/16/20
Straight restart is now implemented.

**Comment by carolyner** [comment]
Dev branch code revised to allow a straight continuation (phase 1) or a "warm start", meaning the algorithm will accept a user-provided sample of model parameters.  The warm start partially addresses phases 2 and 3. The user can modify targets (phase 2) and can revise the parameters (phase 3). Adding parameter would require the user to think about how to add sampled values for this unknown parameter.  This adds functionality but does not fully address the phase 2 goal.

When using the warm start approach, it is important to understand that the sample provided is assumed to be drawn from the prior distribution of model parameters, with uniform priors specified to indicate the prior ranges.  Future enhancements may build from existing simulated targets and weights, so that weighting is back to the originally specified prior distribution.

**Comment by jozik** [comment]
One point for clarification, on this point:

> When using the warm start approach, it is important to understand that the sample provided is assumed to be drawn from the prior distribution of model parameters, with uniform priors specified to indicate the prior ranges.

Is the point here that the uniform priors are used **just for the prior ranges**? That is, we'd assume that the warm start is a sample from, e.g., a posterior generated previously, but that the parameter ranges are needed to constrain the parameter space, and we only use the prior ranges to do the latter. Or am I misunderstanding the intent of the warm start and prior specifications?

**Comment by carolyner** [comment]
@jozik I have edited my comment: it was wrong. The use of warm start is complicated by the need to be able to calculate prior densities for all sampled parameters. Warm start is best viewed as an exploration tool and a step toward our phase 2 goal of being able to add/revise targets.

**Comment by jozik** [comment]
@carolyner thanks! Could the empirical densities be used directly, either easily now or with some modifications later?

**Comment by carolyner** [comment]
@jozik No, the empirical densities cannot be used directly and we don't have plans to include them. Phase 2 would use the mixture distribution in combination with the original prior distributions. An alternative to using the warm start would be to use samples to estimate marginal truncated normal (TN) distributions for each parameter and then start the algorithm with these informative priors, though this approach would not be able to capture prior correlation among parameters or multi-modal priors.
